### PR TITLE
Custom parameters for superadmin creator

### PR DESCRIPTION
**Pull request by nathanchrs**
Implements #170.

Adds configuration for Jophiel's superadmin creator which runs at Jophiel startup.
New options: `superadminCreator.initialPassword`.

Also modified Ansible deploy script so that the Jophiel conf uses the newly added options (using Ansible conf variable `jophiel_superadminCreator_initialPassword`.

### CODE REVIEW

File: deployment/ansible/dist-example/env.yml [hunk 1]
```diff
@@ -83,6 +83,8 @@ jophiel_userResetPassword_requestEmailTemplate_body: xxx
 jophiel_userResetPassword_resetEmailTemplate_subject: xxx
 jophiel_userResetPassword_resetEmailTemplate_body: xxx
 
+jophiel_superadmin_initial_password: xxx
```
Comment (fushar) [inline deployment/ansible/dist-example/env.yml]:
```suggestion
jophiel_superadmin_initialPassword: xxx
```

File: judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@JsonDeserialize(as = ImmutableSuperadminCreatorConfiguration.class)
+public interface SuperadminCreatorConfiguration {
+    SuperadminCreatorConfiguration DEFAULT = new SuperadminCreatorConfiguration.Builder()
+            .enabled(true)
+            .username("superadmin")
+            .initialPassword("superadmin")
+            .initialEmail("superadmin@jophiel.judgels")
+            .build();
+
+    boolean getEnabled();
```
Comment (fushar) [inline judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java]: I think this should be always enabled, so no need for this config

File: judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    Optional<String> getUsername();
```
Comment (fushar) [inline judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java]: Let's always use `superadmin` as username for simplicity. We never need to have another username.

File: deployment/ansible/conf/jophiel.yml.j2 [hunk 4]
```diff
@@ -105,3 +105,9 @@ jophiel:
       subject: UNUSED
       body: UNUSED
 {% endif %}
+
+  superadminCreator:
+    enabled: true
+    username: superadmin
+    initialPassword: {{ jophiel_superadmin_initial_password }}
```
Comment (fushar) [inline deployment/ansible/conf/jophiel.yml.j2]:
```suggestion
    initialPassword: {{ jophiel_superadmin_initialPassword }}
```

**Review by fushar — CHANGES_REQUESTED** [review]

File: deployment/ansible/dist-example/env.yml (hunk 1 above)
Comment (nathanchrs) [reply deployment/ansible/dist-example/env.yml]: OK

File: deployment/ansible/conf/jophiel.yml.j2 (hunk 4 above)
Comment (nathanchrs) [reply deployment/ansible/conf/jophiel.yml.j2]: OK

Comment (nathanchrs) [reply deployment/ansible/conf/jophiel.yml.j2]: Changed to `jophiel_superadminCreator_initialPassword` for consistency with actual config key

File: judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java (hunk 3 above)
Comment (nathanchrs) [reply judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java]: OK

File: judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java (hunk 2 above)
Comment (nathanchrs) [reply judgels-backends/jophiel/jophiel-app/src/main/java/judgels/jophiel/user/superadmin/SuperadminCreatorConfiguration.java]: OK

File: deployment/ansible/dist-example/env.yml (hunk 1 above)
Comment (nathanchrs) [reply deployment/ansible/dist-example/env.yml]: Changed to `jophiel_superadminCreator_initialPassword` for consistency with actual config key

**Review by fushar — APPROVED** [review]

**Pull request merged**